sporkfish/searcher/minimax.py

```diff
@@ -59,34 +62,64 @@ def __init__(
             logging.info("Disabled transposition table in search.")
 
         self._evaluator = evaluator
-        self._move_order = move_order
+
+        # Killer move table - storing quiet beta-cut off moves
+        self._killer_moves = (
+            [
+                [chess.Move.null(), chess.Move.null()]
+                for _ in range(self._searcher_config.max_depth + 1)
+            ]
+            if self._searcher_config.move_order_mode == MoveOrderMode.KILLER_MOVE
+            or self._searcher_config.move_order_mode == MoveOrderMode.COMPOSITE
+            else None
+        )
 
     @property
     def evaluator(self) -> Evaluator:
         return self._evaluator
 
-    def _ordered_moves(self, board: Board, legal_moves: Any, depth: int) -> Any:
+    def _build_move_order_heuristic(
+        self, board: Board, depth: int
+    ) -> MoveOrderHeuristic:
         """
-        Order the given legal moves from best to worst based on a move ordering heuristic.
-
-        This method sorts the legal moves based on a move ordering heuristic, which aims to prioritize
-        moves that are likely to lead to better positions or outcomes. The ordering can help improve
-        the efficiency of the search algorithm by considering more promising moves first.
+        Build and return an instance of MoveOrderHeuristic based on the specified order type.
 
         :param board: The current state of the chess board.
-        :type board: chess.Board
-        :param legal_moves: The legal moves to be ordered.
-        :type legal_moves: Any
+        :type board: Board
         :param depth: The depth of the search.
         :type depth: int
-        :return: The ordered legal moves.
-        :rtype: Any
+        :return: An instance of MoveOrderHeuristic.
+        :rtype: MoveOrderHeuristic
+        :raises TypeError: If the specified order type is not supported.
         """
-        return sorted(
-            legal_moves,
-            key=lambda move: (self._move_order.evaluate(board, move, depth),),
-            reverse=True,
-        )
+        order_type = self._searcher_config.move_order_mode
+        if order_type is MoveOrderMode.MVV_LVA:
+            return MvvLvaHeuristic(board)
+        elif order_type is MoveOrderMode.KILLER_MOVE:
+            return KillerMoveHeuristic(board, self._killer_moves, depth)  # type: ignore
+        elif order_type is MoveOrderMode.COMPOSITE:
+            return CompositeHeuristic(board, self._killer_moves, depth)  # type: ignore
+        else:
+            raise TypeError(
+                f"MoveOrderingHeuristic does not support the creation of MoveOrdering type: \
+                {type(order_type).__name__}."
+            )
+
+    def _update_killer_moves(self, move: chess.Move, depth: int) -> None:
```

Comment:
I personally would prefer having a separate `MoveOrderFactory` class that contains all these move ordering related methods and attributes, and then `MinimaxVariant` can inherit from it. This keeps things more modular and logically separated.

Thoughts?

Not blocker - can always refactor later - I think the searcher classes/functions (`minimaxvariant` and `negamax`) would benefit from some refactoring

Reply:
I didn't consider it in this PR because when we move to Transposition tables, how do you categorise this? Is it a move order method or is it a searcher method? It isn't clear to me. 

I have no objections to moving to a MoveOrderFactory on the condition we clarify what a "move ordering" method/attribute is.

Reply:
Ahh that is good point - didn't think of that (more like I didn't know there is a TT based move ordering tbh). 

This needs some further thinking from my side sorry. In favour of just raising an issue and we can continue the discussion there.

What we have currently is good and should go in. Further design/refactoring can come later - again, keeping ourselves [agile](https://en.wikipedia.org/wiki/Agile_software_development) :) (never really understood what that means, but it sounds fancy)

Reply:
Raised #121